src/arch/x86_64/kernel/start.rs

```diff
@@ -33,6 +36,7 @@ pub unsafe extern "C" fn _start(boot_info: &'static mut BootInfo) -> ! {
 }
 
 unsafe fn pre_init(boot_info: &'static mut BootInfo) -> ! {
+	assert_eq!(boot_info.magic_number, BOOT_IMAGE_MAGIC_NUMBER);
```

Comment:
Since the Unit-tests have garbage in boot_info, I thought it would be reasonable to add an assertion here that checks the magic number.